test/ably/resthttp_test.py

```diff
@@ -135,3 +139,33 @@ def test_custom_http_timeouts(self):
         self.assertEqual(ably.http.http_open_timeout, 8)
         self.assertEqual(ably.http.http_max_retry_count, 6)
         self.assertEqual(ably.http.http_max_retry_duration, 20)
+
+    def test_version_and_lib_in_headers(self):
+        port = 8887
+        def start_server():
+            class Handler(tornado.web.RequestHandler):
+                def get(self):
+                    self.write(dict(self.request.headers))
+
+            app = tornado.web.Application([
+                (r"/", Handler),
+            ])
+            app.listen(port)
+            tornado.ioloop.IOLoop.current().start()
+
+        proc = Process(target=start_server)
+        proc.start()
+        time.sleep(1)
+        ably = AblyRest(token="foo", rest_host="localhost", port=port, tls=False)
+        try:
+            body = ably.http.get("/", skip_auth=True).json()
+        except:
+            raise
+        finally:
+            proc.terminate()
+
+        self.assertIn("X-Ably-Version", body)
+        self.assertEqual(body["X-Ably-Version"], ably.api_version)
+        self.assertIn("X-Ably-Lib", body)
+        self.assertEqual(body["X-Ably-Lib"], ably.lib_variant + ably.api_lib)
```

Comment:
I believe we should be more explicit here in this test with what value you're expecting.  If you are only checking that it matches `lib_variant` + `api_lib`, and someone introduces a regression where `lib_variant` is an empty string, this will still pass. The point of the tests is to ensure we match the spec and prevent regressions.  I don't believe this achieves that in full unfortunately.